src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,223 @@
+                                               horizontalTitleStyle,
+                                               buttonGroupClassName,
+                                               backButtonClassName,
+                                               resetButtonClassName,
+                                               skipButtonClassName,
+                                               continueButtonClassName,
+                                               alternative=false,
+                                               verticalStepper=false,
+                                               stepsArr,
+                                               detectError,
+                                               detectOptional,
+                                               customErrorSvg,
+                                               customFinishedSvg,
+                                               children}) => {
+
+    const [activeStep, setActiveStep] = useState<number>(0);
+    const [skipped, setSkipped] = useState(new Set<number>());
+    const [errorSteps, setErrorSteps] = useState(new Set<number>());
+
+
+    const checkIsOptional = (step: number): boolean => {
+        if (detectOptional) {
+            return detectOptional(step);
+        } else {
+            return false;
+        }
+    }
+
+    const checkIsError = (step: number) => {
+        if (detectError) {
+            return detectError(step);
+        } else {
+            return false;
+        }
+    }
+
+    const checkIsSKipped = (step: number) => {
+        return skipped.has(step);
+    }
+
+    const handleSkipClick = () => {
+        if (!checkIsOptional(activeStep)) {
+            throw new Error('You can\'t skip a step that is not optional.')
+        }
+        setActiveStep(prevState => prevState + 1);
+        setSkipped(prevState => {
+            const newSkipped = new Set(prevState.values())
+            newSkipped.add(activeStep);
+            return newSkipped;
+        })
+    }
+
+    const handleBackClick = () => {
+        setActiveStep(prevState => prevState - 1);
+    }
+
+    const handleNextClick = (step: number) => {
+        const isError = checkIsError(step);
+        if (!isError) {
+            setActiveStep(prevState => prevState + 1);
+        } else {
+            setErrorSteps(errorSteps.add(step));
+            forceReRender();
+        }
+    }
+
+    const handleResetClick = () => {
+        setSkipped(new Set());
+        setErrorSteps(new Set());
+        setActiveStep(0);
+    }
+
+
+    let verticalStyle = classNames('stepper-container--vertical');
+    let horizontalStyle = classNames('stepper-container--horizontal');
+    let horizontalItemStyle = classNames('stepper-item-container--horizontal');
+    let horizontalContentStyle = classNames('stepper-content-container--horizontal');
+
+    verticalStyle = verticalContainerClassName ? verticalStyle + ' ' + verticalContainerClassName : verticalStyle;
```

Comment:
You don't have to call the function classNames() if you just want to pass one class name. I think it would be better if you combine these lines with the code above. For example,`let verticalStyle = classNames('stepper-container--vertical', { [verticalContainerClassName]: verticalContainerClassName !== undefined });`